Declaration in @NgModule duplicating when generating stuff

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug report

### Command (mark with an `x`)
<!-- Can you pin-point the command or commands that are effected by this bug? -->
<!-- ✍️edit: -->
```
- [x] generate
```

### Is this a regression?

Probably, found it on version 8.3.5

### Description

The "declaration" key in @NgModule duplicating when generating stuff.

## 🔬 Minimal Reproduction

1. Put a quotes or a double-quotes on the "declaration" key at any @NgModule. Or set "object-literal-key-quotes" on tslint to [true, "always"]. Originally, it would look like this:
<pre><code>@NgModule({
  declarations: [Component1, Component2],
  imports: [
    CommonModule,
    OtherModule
  ]
})</code></pre>
2. Run ng lint --fix (if you did the tslint). It should now look like this:
<pre><code>@NgModule({
  "declarations": [Component1, Component2], // ---> added quotes on the "declaration"
  "imports": [
    CommonModule,
    OtherModule
  ]
})</code></pre>

3. Generate a component where it would import on the said module. It either generates another "declaration" key with or without quotes (depending on your schematics whether you run lint after generating). It would now look like this:
<pre><code>@NgModule({
  "declarations": [Component1, Component2],
  "imports": [
    CommonModule,
    OtherModule
  ]
  "declarations": [Component3] // ---> It would generate this with or without quotes instead of merging it with the existing one
})</code></pre>

## 🔥 Exception or Error

Running ng serve now throws the error

`error TS2300: Duplicate identifier 'declarations'.`

## Other Notes

Not big of an issue, but it kills some of the tasks that should be automated when you have a custom tslint settings at the same time.

--- a/packages/schematics/angular/utility/ast-utils.ts
+++ b/packages/schematics/angular/utility/ast-utils.ts
@@ -377,8 +377,7 @@ export function getMetadataField(
     // Filter out every fields that's not "metadataField". Also handles string literals
     // (but not expressions).
     .filter(({ name }) => {
-      return (ts.isIdentifier(name) || ts.isStringLiteral(name))
-        && name.getText() === metadataField;
+      return (ts.isIdentifier(name) || ts.isStringLiteral(name)) && name.text === metadataField;
     });
 }
 

--- a/packages/schematics/angular/utility/ast-utils_spec.ts
+++ b/packages/schematics/angular/utility/ast-utils_spec.ts
@@ -91,6 +91,29 @@ describe('ast utils', () => {
     expect(output).toMatch(/declarations: \[\nAppComponent,\nFooComponent\n\]/);
   });
 
+  it('should add declarations to module when PropertyAssignment is StringLiteral', () => {
+    moduleContent = tags.stripIndents`
+    import { BrowserModule } from '@angular/platform-browser';
+    import { NgModule } from '@angular/core';
+    import { AppComponent } from './app.component';
+
+    @NgModule({
+      "declarations": [
+        AppComponent
+      ],
+      "imports": [
+        BrowserModule
+      ],
+      "providers": [],
+      "bootstrap": [AppComponent]
+    })`;
+    const source = getTsSource(modulePath, moduleContent);
+    const changes = addDeclarationToModule(source, modulePath, 'FooComponent', './foo.component');
+    const output = applyChanges(modulePath, moduleContent, changes);
+    expect(output).toMatch(/import { FooComponent } from '.\/foo.component';/);
+    expect(output).toMatch(/"declarations": \[\nAppComponent,\nFooComponent\n\]/);
+  });
+
   it('should add metadata', () => {
     const source = getTsSource(modulePath, moduleContent);
     const changes = addSymbolToNgModuleMetadata(source, modulePath, 'imports', 'HelloWorld');

